### PR TITLE
db: remove wal disabling / fast-and-loose option.

### LINKED ISSUE #8779: Change name/description of fast_and_loose.
The config option fast-and-loose has a name and documentation that currently implies that the node can be restarted, it just isn't resilient to failure.  This makes it sound like a great solution for running a dev chain or test chain, and lead me (and others #7910) to try to use it that way only to find out after much troubleshooting that that isn't the case.

My particular use case is to create a Parity docker image that has some base contracts pre-delpoyed to it that I can then test against.  For example, one might create a Dockerfile like:
```Dockerfile
FROM parity/parity:v1.11.1

WORKDIR /
COPY dev-key.json /parity/keys/DevChain/dev-key.json
COPY chain-spec.json /parity/chain-spec.json
COPY config.toml /parity/config.toml
RUN echo "" > /parity/password

COPY deploy-contracts.sh /app/deploy-contracts.sh
RUN chomd 555 /app/deploy-contracts.sh
RUN /parity/parity --config /parity/config.toml \
    && /app/deploy-contracts.sh \
    && kill -TERM %1

ENTRYPOINT [ "/parity/parity", "--config", "/parity/config.toml" ]
```

I would expect fast-and-lose to generally work here because I stop Parity once cleanly after deploying contracts, then when the container is run in the future it will start from that state (with contracts deployed).  However, with fast-and-loose in `config.toml`, this doesn't actually work.

The poor naming/documentation around this feature resulted in notable engineering time spent debugging.

Side note: It would be preferable if there were a mode that actually did what this implied, and supported clean exits but made no guarantees about unclean exits.  That would allow the above use-case to actually work.

### PR DESCRIPTION
**Pull request by 5chdn**
fix #8779

ref #1765

### CODE REVIEW

**Comment by andresilva** [comment]
I'd actually say we remove it since this is really unsafe and will just give us trouble (more issues with database corruption). It's better to just provide a bigger cache size for performance.

**Comment by 5chdn** [comment]
Hold on.

File: parity/cli/mod.rs [hunk 1]
```diff
@@ -883,6 +879,10 @@ usage! {
 			"--warp",
 			"Does nothing; warp sync is enabled by default. Use --no-warp to disable.",
 
+			FLAG flag_fast_and_loose: (bool) = false, or |_| None,
+			"--fast-and-loose",
+			"Does nothing. DB WAL is always activated.",
```
Comment (andresilva) [inline parity/cli/mod.rs]: Maybe mention here that the option is deprecated?

Comment (5chdn) [reply parity/cli/mod.rs]: This is just a placeholder really since `--help` does not print these options anymore.

**Comment by 5chdn** [comment]
@andresilva done :blush:

**Comment by 5chdn** [comment]
- [ ] needs proper deprecated handling, ref #9036

**Comment by 5chdn** [comment]
ok, added to deprecated::does_nothing() :+1:

File: parity/deprecated.rs [hunk 2]
```diff
@@ -203,6 +203,12 @@ pub fn find_deprecated(args: &Args) -> Vec<Deprecated> {
 		result.push(Deprecated::Removed("--tx-queue-ban-time"));
 	}
 
+	// Removed in 2.0.
+
+	if args.flag_fast_and_loose {
+		result.push(Deprecated::DoesNothing("--fast-and-loose"));
```
Comment (andresilva) [inline parity/deprecated.rs]: Minor nit, but why `DoesNothing` instead of `Removed`?

Comment (5chdn) [reply parity/deprecated.rs]: Correct. It should be removed.

**Comment by 5chdn** [comment]
FFFFFFFFFFFUUU... :rofl:

**Comment by dvdplm** [comment]
**NOTE** We should merge https://github.com/paritytech/parity-common/pull/7 before merging this. If you came here to review, please consider reviewing that one instead.

**Pull request merged**